knox/crypto.py

```diff
@@ -24,5 +24,5 @@ def hash_token(token):
     exception will be raised
     '''
     digest = hashes.Hash(sha(), backend=default_backend())
-    digest.update(binascii.unhexlify(token))
+    digest.update(binascii.unhexlify(binascii.hexlify(bytes(token, 'utf-8'))))
```

Comment:
this is a bit weird, but ensures that the token is always hex-compatible.

Let me know, if you got a better solution

Reply:
I don't have a better solution, but that's sufficiently weird that I think it would benefit from being put in its own method,  some name like `make_hex_compatible` - otherwise someone will look at that 6 months from now and be left scratching their head as to why

Reply:
Good point!